integrations/pagerduty/clients/pagerduty.py

```diff
@@ -131,3 +137,50 @@ async def create_webhooks_if_not_exists(self) -> None:
                 logger.error(
                     f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
                 )
+
+    async def get_oncall_user(self, escalation_policy_id: str) -> dict[str, Any]:
+        url = f"{self.api_url}/oncalls"
+
+        async with httpx.AsyncClient() as client:
+            try:
+                response = await client.get(
+                    url,
+                    params={
+                        "escalation_policy_ids[]": escalation_policy_id,
+                        "include[]": "users",
+                    },
+                    headers=self.api_auth_header,
+                )
+                response.raise_for_status()
+                data = response.json()
+                return data
+            except httpx.HTTPStatusError as e:
+                logger.error(
+                    f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+                )
+                raise
+
+    async def update_oncall_users(
+        self, data: dict[str, Any], data_key: str
+    ) -> list[Any]:
+        logger.info("Fetching and matching who is on-call for services")
+        escalation_policy_ids = [
+            service["escalation_policy"]["id"] for service in data[data_key]
+        ]
+
+        oncall_users = await self.get_oncall_user(",".join(escalation_policy_ids))
+        all_data = []
+
+        for service in data[data_key]:
+            escalation_policy_id = service["escalation_policy"]["id"]
+
```

Comment:
What will happen if my service doesn;t have an escalation policy?

Reply:
From the UI, it appears that the escalation policy is always **required** when creating a service
![image](https://github.com/port-labs/Port-Ocean/assets/15999660/d04ad2f6-25c7-47a0-a06a-0ea484ce694c)

Also, from the [Create Service API](https://developer.pagerduty.com/api-reference/7062f2631b397-create-a-service), the escalation_policy is a required property